File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var maxScore = function(s) {
    let ones = 0, zeros = 0, res = 0;

    s.split('').forEach(ele => {
        if(ele === '1') {
            ones++
        }
    });

    for(var i = 0; i < s.length - 1; i++) {
        if(s[i] == '1') {
            ones--;
        } else {
            zeros++;
        }

        res = Math.max(res, ones + zeros);
    }

    return res;
};

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int zeros = 0, ones = 0;
        int n = s.length();
        for(int i = 1; i < n; i++) {
            if(s[i] == '1') {
                ones++;
            }
        }

        if(s[0] == '0') {
            zeros++;
        }

        int res = ones + zeros;

        int j = 2;

        while(j < n) {
            if(s[j - 1] == '0') {
                zeros++;
            } else {
                ones--;
            }

            res = max(res, ones + zeros);
            j++;
        }

        return res;
    }
};